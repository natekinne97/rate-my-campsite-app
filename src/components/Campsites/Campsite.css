/*all posts*/
.all-posts{
	/* display: flex;
  	justify-content: space-evenly;
  	flex-wrap: wrap; */
	width: 90%;
	display: flex;
	justify-content: space-evenly;
	flex-wrap: wrap;
	align-items: center;
	margin: 0 auto;
	margin-top: 2%;
}

/* post the container holding the pic and info */
.post{
	
	color: rgb(0, 0, 0, 0.7);
	cursor: pointer;
	text-align: center;
	margin: 1%;
	margin-bottom: 8%;
	height: 52vh;
	width: 39vh;
	/* card style box shadow */
	box-shadow: 0 2px 1px -1px rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 1px 3px 0 rgba(0,0,0,.12);
}

.paragraphs{
	padding: 7%;
}

.post p{
	margin: 0;
}


.img-post{
	width: 100%;
	height: 40%;
	background-size: cover;
  	background-position: 50% 50%;
}

.tent-rate{
	color: rgb(0, 0, 0, 0.4);
}

.tent{
	font-size: 15px;
}


/* tablet and higher */
@media only screen and (min-width: 768px){
	.post{
		width: 23vh;
	}
}