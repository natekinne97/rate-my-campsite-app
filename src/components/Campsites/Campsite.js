import React from 'react';
import { Link } from 'react-router-dom'
import './Campsite.css';
import STORE from '../../Store';

class Campsite extends React.Component{



    render(){
        return(
            <div className="all-posts">
                {STORE.Campsites.map(site=>(
                    
                    <Link key={site.id}  to={`/info/${site.id}`}>
                        <div className="post">
                            <div><img className="img-post" src={site.picture} alt={site.alt} /></div>
                            <p>{site.state}</p>
                            <p>{site.rating}/{site.reviews}</p>
                        </div>
                    </Link>
                   
                ))}
            </div>
        );
    }
}

export default Campsite;