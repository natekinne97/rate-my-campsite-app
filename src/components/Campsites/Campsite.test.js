import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Campsite from './Campsite';

describe.only('Review form component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Campsite />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Campsite />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});