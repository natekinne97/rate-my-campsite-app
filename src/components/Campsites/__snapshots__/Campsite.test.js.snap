// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review form component renders the UI as expected 1`] = `
<div
  className="all-posts"
/>
`;
